clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1045,6 +1052,7 @@ public void fetchResponseWithUnexpectedPartitionIsIgnored() {
 
         ConsumerRecords<String, String> records = consumer.poll(Duration.ZERO);
         assertEquals(0, records.count());
+        assertThat(records.metadata(), equalTo(emptyMap()));
```

Comment:
I dropped in a couple of extra assertions where it seemed appropriate. There are also some new tests farther down to really exercise the new code.